deepinv/training/run_logger.py

```diff
@@ -0,0 +1,499 @@
+from __future__ import annotations
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from datetime import datetime
+from logging import getLogger
+from pathlib import Path
+from typing import Any
+import json
+import logging
+import os
+import platform
+
+from torchvision.utils import save_image
+import torch
+
+
+def get_timestamp() -> str:
+    """Get current timestamp string.
+
+    :return str: timestamp, with separators determined by system.
+    """
+    # ":" is not allowed on Windows filenames
+    sep = "_" if platform.system() == "Windows" else ":"
+    return datetime.now().strftime(f"%y-%m-%d-%H{sep}%M{sep}%S")
+
+
+@dataclass
+class RunLogger(ABC):
+    """
+    Abstract base class for logging training runs.
+    """
+
+    @abstractmethod
+    def init_logger(self, hyperparams: dict[str, Any] | None = None) -> None:
```

Comment:
you expose the hyperparams argument yet all examples seem to be only using Logger.init_logger(). is there a valid use case for passing a dict here that we can exemplify in some examples? or at least document the high level interface for loggers in the user guide somewhere alongside the trainer docs?